openhcl/underhill_core/src/dma_manager.rs

```diff
@@ -0,0 +1,131 @@
+//use std::sync::{Mutex, Weak};
+
+use memory_range::MemoryRange;
+use std::{collections::HashMap, sync::{Arc, Mutex}};
+use page_pool_alloc::PagePool;
+use user_driver::{memory::MemoryBlock, vfio::VfioDmaBuffer};
+use user_driver::lockmem::LockedMemorySpawner;
+
+#[derive(Clone)]
```

Comment:
you should make this inspectable, and make sure it's wired up to the inspection graph. For things inside LoadedVm they have to be done manually, but you can follow the prior art here.